plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt

```diff
@@ -12,29 +12,37 @@ import com.mapbox.android.core.location.LocationEngineResult
 import com.mapbox.android.core.permissions.PermissionsManager
 import com.mapbox.common.Logger
 import com.mapbox.geojson.Point
+import com.mapbox.maps.plugin.PuckBearingSource
+import com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings
 import java.lang.ref.WeakReference
 import java.util.concurrent.CopyOnWriteArrayList
 
 /**
  * Default Location Provider implementation, it can be overwritten by users.
  */
-internal class LocationProviderImpl(context: Context) :
-  LocationProvider, LocationEngineCallback<LocationEngineResult> {
+internal class LocationProviderImpl(context: Context, var settings: LocationComponentSettings) :
+  LocationProvider,
+  LocationEngineCallback<LocationEngineResult>,
+  LocationCompassEngine.CompassListener {
   private val contextWeekRef: WeakReference<Context> = WeakReference(context)
   private val locationEngine = LocationEngineProvider.getBestLocationEngine(context)
-
+  private val locationCompassEngine = LocationCompassEngine(context)
```

Comment:
could we expose the compass engine separately from the GPS location engine? As we might want to get the compass heading independently from gps course to manipulate camera.

Reply:
I would also expect to have LocationProvider / CompassProvider as a separate entities and some holder object that  aggregates both and handles which one to query

Reply:
There is an interface `LocationComponentPlugin#getLocationProvider` and `locationProvider.registerLocationConsumer`, in this way, we could get the compass heading value for both `HEADING` and `COURSE`